submissions/YuraZagor/tjsw/index.js

```diff
@@ -0,0 +1,166 @@
+const manNamesArr = ['Vasil', 'Yurij', 'Zahar', 'Taras', 'Olexiy', 'Oleg', 'Mykola', 'Sashko', 'Sergij', 'Bohdan'];
+const womanNamesArr = ['Hanna', 'Helena', 'Vasilisa', 'Yulya', 'Oksana', 'Olena', 'Olga', 'Myroslava', 'Svitlana', 'Bohdana'];
+const animalNamesArr = ['Joy', 'Happy', 'Lucky', 'Puffy'];
+const genderArr = ['male', 'female'];
+
+let catSay = 'meau';
+let dogSay = 'woufff';
+let humanSay = 'Glory to Ukraine!';
+
+function randomSelect(arr){
+  return arr[Math.floor(Math.random()*arr.length)]
+};
+
+let manName = randomSelect(manNamesArr);
+let womanName = randomSelect(womanNamesArr);
+let animalName = randomSelect(animalNamesArr);
+
+const inhabitants = [
+  { species: 'human', 
+    gender: 'male',
+    name: 'Vasil',
+    saying: 'Slava natsiyi',
+    friend: 'Hanna',
+    legs: 2,
+    hands: 2,
+  },
+  { species: 'human', 
+    gender: 'female',
+    name: 'Hanna',
+    saying: 'Smert vorogam',
+    friend: 'Vasil',
+    legs: 2,
+    hands: 2,
+  },];
+
+const dogsVolier = [{ 
+  species: 'Dog', 
+  gender: 'female',
+  name: 'Guffy',
+  saying: dogSay,
+  friend: randomSelect(inhabitants).name,
+  paws: 4,
+  tail: 1,
+},];
+
+class Inhabitant {  
+  constructor(species, name, saying, friend, gender, habitat = inhabitants){
+    this.species = species;
+    this.name = name;
+    this.saying = saying
+    this.friend = friend;
+    this.gender = gender;
+    this.habitat = habitat;
+  }
+  accountInhabitants(){
+    this.habitat.push(this);
+  };
+  commonProperties(){
+    const { species, name, saying, friend, gender} = this;
+    return [species, name, saying, friend, gender]
+  }
+};
+
+class Animal extends Inhabitant{
+  constructor( species, name, saying, friend, gender, paws, tail ){
+    super(species, name, saying, friend, gender);
+    this.paws = paws;
+    this.tail = tail;
+  };
+  populate(){
```

Comment:
This method does at least 3 clearly distinctive things:
(1) adds current class' properties to the list of properties of a parent class
(2) prints them
(3) calls `this.accountInhabitants` (no arguments and also says it will add many inhabitant**s**)

Separation of Concerns is still a thing.

Also, this method returns nothing, which means that if a create a similar method in a child class (e.g. add a Cat-specific property) I won't be able to re-use it but rather will be forced to redefine it from the scratch. What a waste of code.

Reply:
1) My bad. 
3) renamed to _registerInItsHabitat_ to better reflect what's going on there. 

Not sure how I can do without 2 functions called. Do you mean I should make separate method for printing, and than call both like: 
populate( ) { printingMethod( ); registerInItsHabitat( ) }; ?  

For the last part I can smell a hint to ...rest parameters, but I'm not sure how they can be bound to this.  Upd. They can, which moved all methods down to class Inhabitant. I like that!